tests/integration/wordpress_client.py

```diff
@@ -192,3 +200,72 @@ def upload_media(self, filename: str, content: bytes, mimetype: str = None) -> t
         if media["source_url"] not in image_urls:
             image_urls.append(media["source_url"])
         return image_urls
+
+    def associate_ubuntu_one(self, username, password):
+        openid_setting_url = f"http://{self.host}/wp-admin/users.php?page=your_openids"
+        openid_setting_page = self._get(openid_setting_url)
+        nonce = re.findall(
+            '<input type="hidden" id="_wpnonce" name="_wpnonce" value="([^"]+)" />',
+            openid_setting_page.text,
+        )[1]
+        openid_redirect = self._post(
+            openid_setting_url,
+            data={
+                'openid_identifier': 'login.ubuntu.com',
+                '_wpnonce': nonce,
+                '_wp_http_referer': '/wp-admin/users.php?page=your_openids',
+                'action': 'add',
+            },
+        )
+        openid_args = re.findall(
+            '<input type="hidden" name="([^"]+)" value="([^"]+)" />', openid_redirect.text
+        )
+        openid_args = dict(openid_args)
+        login_page = self._post(
+            "https://login.ubuntu.com/+openid",
+            data=openid_args,
+        )
+        csrf_token = re.findall(
+            "<input type='hidden' name='csrfmiddlewaretoken' value='([^']+)' />", login_page.text
+        )[0]
+        login_link = re.findall(
+            '<a id="login-link" data-qa-id="login_link" href="([^"]+)" class="p-link--soft">',
+            login_page.text,
+        )[0]
+        login_url = "https://login.ubuntu.com" + login_link
+        confirm_page = self._post(
+            login_url,
+            data={
+                'csrfmiddlewaretoken': csrf_token,
+                'email': username,
+                'user-intentions': 'login',
+                'password': password,
+                'continue': '',
+                'openid.usernamesecret': '',
+            },
+            headers={"Referer": login_page.url},
+        )
+        csrf_token = re.findall(
+            "<input type='hidden' name='csrfmiddlewaretoken' value='([^']+)' />", confirm_page.text
+        )[0]
+        self._post(
+            confirm_page.url,
+            data={
+                'csrfmiddlewaretoken': csrf_token,
+                'nickname': 'on',
+                'email': 'on',
+                'fullname': 'on',
+                'ok': '',
+                'yes': '',
+                'openid.usernamesecret': '',
+            },
+            headers={"Referer": confirm_page.url},
+            except_status_code=200,
+        )
+
+    def list_associated_ubuntu_one_accounts(self):
```

Comment:
We have a new standard for docstrings, do you think it applies here? https://github.com/canonical/is-charms-contributing-guide/blob/main/CONTRIBUTING.md#docstrings

We have a new standard for type hints, do you think it applies here? https://github.com/canonical/is-charms-contributing-guide/blob/main/CONTRIBUTING.md#type-hints

Reply:
Fixed, thanks!